p2p/src/net/mod.rs

```diff
@@ -88,12 +89,35 @@ pub trait NetworkingService {
         shutdown: Arc<SeqCstAtomicBool>,
         shutdown_receiver: oneshot::Receiver<()>,
         subscribers_receiver: mpsc::UnboundedReceiver<P2pEventHandler>,
-    ) -> crate::Result<(
-        Self::ConnectivityHandle,
-        Self::MessagingHandle,
-        Self::SyncingEventReceiver,
-        JoinHandle<()>,
-    )>;
+    ) -> crate::Result<
+        NetworkingServiceJoinHandle<
+            Self::ConnectivityHandle,
+            Self::MessagingHandle,
+            Self::SyncingEventReceiver,
+        >,
+    >;
+}
+
+#[derive(Debug)]
+pub struct NetworkingServiceJoinHandle<ConnectivityHandle, MessagingHandle, SyncingEventReceiver>
+where
+    SyncingEventReceiver: Send,
+    MessagingHandle: Send + Sync + Clone,
```

Comment:
I don't think Clone makes sense here. It's not needed.

Reply:
Btw, you didn't have to create another PR to remove this Clone. I only meant that in this struct Clone doesn't make sense. But anyway, I'm OK with that either way.